### PR TITLE
feat: add info controller

### LINKED ISSUE #480: Let LAPIS return an info object alongside the data
Currently each endpoint returns an object, similar to `{"data": "someData"}`, where someData can be the results of an aggregated, mutation or details request. 

The returned object should be extended, such that it also contains an Info object, similar to

`{"data": "someData, "info": {"dataVersion": "1234"}`

The info object should contain for now only the dataVersion, which is also returned in the response header.
The info object should only be added if the request does not fail.

### PR DESCRIPTION
**Pull request by JonasKellerer**
Resolves #480

### CODE REVIEW

File: lapis2/src/main/kotlin/org/genspectrum/lapis/silo/SiloClient.kt [hunk 1]
```diff
@@ -72,18 +87,37 @@ class SiloClient(
...
+            .GET()
+            .build()
+
+        val response = httpResponse(client, request)
 
         try {
-            return objectMapper.readValue(response.body(), query.action.typeReference).queryResult
+            val infoData = objectMapper.readValue(response.body(), InfoData::class.java)
+            val dataVersion = getDataVersion(response)
+            return infoData.copy(dataVersion = dataVersion)
```
[the remark this answers is not in the document]
Comment (JonasKellerer) [reply lapis2/src/main/kotlin/org/genspectrum/lapis/silo/SiloClient.kt:106]: Dont we want to get the sequenceCount as well from SILO? Or do we remove that from InfoData?

[the remark this answers is not in the document]
Comment (fengelniederhammer) [reply lapis2/src/main/kotlin/org/genspectrum/lapis/silo/SiloClient.kt:106]: I would remove it here.

File: lapis2/src/main/kotlin/org/genspectrum/lapis/controller/ControllerDescriptions.kt [hunk 2]
```diff
@@ -41,3 +41,5 @@ const val ALIGNED_SINGLE_SEGMENTED_NUCLEOTIDE_SEQUENCE_ENDPOINT_DESCRIPTION =
 const val ALIGNED_MULTI_SEGMENTED_NUCLEOTIDE_SEQUENCE_ENDPOINT_DESCRIPTION =
     "Returns a string of fasta formatted aligned nucleotide sequences of the requested segment. " +
         "Only sequences matching the specified sequence filters are considered."
+
+const val INFO_ENDPOINT_DESCRIPTION = "Returns information about the Lapis server."
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/controller/ControllerDescriptions.kt:45]: This is unused now?

File: lapis2/src/main/kotlin/org/genspectrum/lapis/openApi/OpenApiDocs.kt [hunk 3]
```diff
@@ -209,7 +209,8 @@ fun buildOpenApiSchema(
                 )
                 .addSchemas(LIMIT_SCHEMA, limitSchema())
                 .addSchemas(OFFSET_SCHEMA, offsetSchema())
-                .addSchemas(FORMAT_SCHEMA, formatSchema()),
+                .addSchemas(FORMAT_SCHEMA, formatSchema())
+                .addSchemas(INFO_RESPONSE_SCHEMA, infoResponseSchema()),
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/openApi/OpenApiDocs.kt:213]: Do we still need this line?

File: lapis2/src/main/kotlin/org/genspectrum/lapis/openApi/Schemas.kt [hunk 4]
```diff
@@ -59,7 +60,13 @@ const val FORMAT_SCHEMA = "DataFormat"
 const val FIELDS_TO_AGGREGATE_BY_SCHEMA = "FieldsToAggregateBy"
 const val DETAILS_FIELDS_SCHEMA = "DetailsFields"
 
+const val LAPIS_INFO_DESCRIPTION = "Information about LAPIS."
+const val LAPIS_DATA_VERSION_EXAMPLE = "1702305399"
 const val LAPIS_DATA_VERSION_DESCRIPTION = "The data version of data in SILO."
+const val LAPIS_DATA_VERSION_HEADER_DESCRIPTION = "$LAPIS_DATA_VERSION_DESCRIPTION " +
+    "Same as the value returned in the info object in the response body."
+const val LAPIS_DATA_VERSION_RESPONSE_DESCRIPTION = "$LAPIS_DATA_VERSION_DESCRIPTION " +
+    "Same as the value returned in the info object in the response body."
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/openApi/Schemas.kt:69]:
```suggestion
    "Same as the value returned in the info object in the header '$LAPIS_DATA_VERSION_HEADER'."
```

File: lapis2/src/main/kotlin/org/genspectrum/lapis/silo/SiloClient.kt [hunk 5]
```diff
@@ -70,18 +85,30 @@ class SiloClient(
...
-        } catch (exception: Exception) {
-            val message = "Could not parse response from silo: " + exception::class.toString() + " " + exception.message
-            throw RuntimeException(message, exception)
-        }
+    fun callInfo(): InfoData {
+        log.info { "Calling SILO info" }
+
+        val client = HttpClient.newHttpClient()
+        val request = HttpRequest.newBuilder(URI("$siloUrl/info"))
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/silo/SiloClient.kt:97]: We don't need the header. There is no content :)

**Review by fengelniederhammer — APPROVED** [review]

File: lapis2/src/main/kotlin/org/genspectrum/lapis/openApi/Schemas.kt (hunk 4 above)
Comment (JonasKellerer) [reply lapis2/src/main/kotlin/org/genspectrum/lapis/openApi/Schemas.kt:69]: ... copilot hat da wohl nicht das geschrieben, was es sollte ... :see_no_evil:

**Pull request merged**